pkg/transform/kubernetes/kubernetes.go

```diff
@@ -289,6 +289,53 @@ func populateContainerHealth(app *spec.App) error {
 	return nil
 }
 
+func populateEnvFrom(app *spec.App) error {
+	// iterate on the containers so that we can extract envFrom
+	// that we have custom defined
+	for ci, c := range app.Containers {
+		for ei, e := range c.EnvFrom {
+			cmName := e.ConfigMapRef.Name
+
+			// we also need to check if the configMap specified if it exists
+			var cmFound bool
+			// to populate the envs we also need to know the data
+			// from configmaps defined in the app
+			for _, cm := range app.ConfigMaps {
+				// we will only populate the configMap that is specified
+				// not every configMap out there
+				if cm.Name != cmName {
+					continue
+				}
+				// start populating
+				for k, _ := range cm.Data {
+					// here we are directly referring to the containers
+					// from app.PodSpec.Containers because that is where data
+					// is parsed into so populating that is more valid thing to do
+					app.PodSpec.Containers[ci].Env = append(app.PodSpec.Containers[ci].Env, api_v1.EnvVar{
```

Comment:
`app.Container` has data that is parsed from file, but it is copied into `app.PodSpec.Container` and it is maintained after that. Our ultimate aim here is to put all data from various sources into `app.PodSpec` so that we can give the `app.PodSpec` to any controller without having to worry if all data from various sources is properly copied into places or not. For more details maybe see this issue https://github.com/kedgeproject/kedge/issues/98 where I have tried to explain it.